embrace-android-sdk/src/test/java/io/embrace/android/embracesdk/ndk/EmbraceNdkServiceTest.kt

```diff
@@ -55,53 +59,12 @@ import java.util.concurrent.ExecutorService
 internal class EmbraceNdkServiceTest {
```

Comment:
Not resetting the mocks in this class made it so that certain settings were dependent on state set outside of the test, which isn't great. Rewrote this so things are reset at each iteration so tests will also start from a clean state so will run deterministically irrespective of execution order.